apps/studio/src/server/modules/folder/folder.router.ts

```diff
@@ -1,24 +1,57 @@
+/* eslint-disable import/prefer-default-export */
 import { protectedProcedure, router } from '~/server/trpc'
-import {
-  createFolderSchema,
-  readFolderOrTopLevelFolderSchema,
-} from '~/schemas/folder'
+import { createFolderSchema, readFolderSchema } from '~/schemas/folder'
 
 export const folderRouter = router({
   create: protectedProcedure
     .input(createFolderSchema)
     .mutation(async ({ ctx, input }) => {
       return { folderId: '' }
     }),
-  readFolderOrTopLevelFolder: protectedProcedure
-    .input(readFolderOrTopLevelFolderSchema)
-    .query(async ({ input, ctx }) => {
-      // TODO: Fill these in later
-      const folderName: string = ''
-      const children: { id: string; name: string; type: 'page' | 'folder' }[] =
-        []
-      // Not sure if a backpointer is needed here
-      const parentId: string = ''
+  readFolder: protectedProcedure
+    .input(readFolderSchema)
+    .query(async ({ ctx, input }) => {
+      // Things that aren't working yet:
+      // 0. Perm checking
+      // 1. Last Edited user and time
+      // 2. Page status(draft, published)
+
+      const folderResult = await ctx.db
+        .selectFrom('Resource')
+        .select(['name', 'parentId'])
+        .where('id', '=', input.resourceId)
+        .executeTakeFirstOrThrow()
+      const childrenResult = await ctx.db
+        .selectFrom('Resource')
+        .select(['id', 'name', 'blobId'])
+        .where('parentId', '=', input.resourceId)
+        .execute()
+      const children = childrenResult.map((c) => {
+        if (c.blobId) {
+          return {
+            id: c.id,
+            name: c.name,
+            type: 'page',
+            lastEditDate: new Date(0),
+            lastEditUser: 'Coming Soon',
+            permalink: '/placeholder',
+            status: 'published',
+          }
+        }
+        return {
+          id: c.id,
+          name: c.name,
+          type: 'folder',
+          lastEditDate: 'folder',
+          lastEditUser: 'Coming Soon',
+          permalink: '/placeholder',
+          status: 'folder',
+        }
+      })
```

Comment:
this seems to be slightly off to me - `lastEditDate/status` for folders isn't very useful.

Reply:
thanks for pointing this out. realise they can be different